packages/common/src/hooks/useSearchParameters/useSearchParameters.ts

```diff
@@ -0,0 +1,26 @@
+import { useSearchParams } from 'react-router-dom';
+
+interface SearchParams {
+  get: (key: string) => string | null;
+  set: (params: Record<string, string>) => void;
+  getNumber: (key: string) => number;
+}
+
+export const useSearchParameters = (): SearchParams => {
```

Comment:
popped this in because:
- the `set` method on `searchParams` didn't do anything
- the `setSearchParams` replaces search params with the object passed to it, which isn't helpful if you want to update one parameter only and retain others

Reply:
cool, I like!